src/style-guide.scrbl

```diff
@@ -188,6 +190,43 @@ fun sqrt(n :: Number%(non-negative)) -> Number:
 end
 }
 
+@subsection{Parametric Annotations}
+
+You should avoid using @tt{Any} as the subtype of a parametric data structure,
+like a @tt{List}.
+
+In general, a @tt{List<Any>} definition is often undesirable because it
+indicates elements of any type can be members of the list. In particular,
+this subtly implies that multiple elements of different types could exist.
+In other words, @tt{[list: 1, "apple", true]} would satisfy the annotation
+@tt{List<Any>}. Creating @tt{List}s like this is usually bad practice.
+
+In Pyret, we can do this by parametrizing the type. Rather than:
+
+@codedisp{
+fun search(lst :: List<Any>) -> Any: ... end
+}
+
+We should instead do something like:
+
+@codedisp{
+fun search<A>(lst :: List<A>) -> A: ... end
+}
+
+In the code above, we don't know or care what the type the elements of 
+@tt{lst} are, but we are asserting that they all be of the same type—namely 
+@tt{A}. The use of @tt{A} as the return type also means that the type of the 
+value returned by @tt{search} must also be the same as the type of the
+elements of @tt{list}. 
+
+In other words, using parameterized types allows us to make the subtle but
+important distinction between "any type at all" (@tt{List<Any>}) and
+"any single type, for any particular list" (@tt{List<A>}).
+
+Note that the actual name of the type parameter is arbitrary (we could have 
```

Comment:
_a_ convention?

Also, I believe recently @jpolitz just changed the documentation for `fold` so that type variables are not a single letter, which might invalidate this paragraph.